pkg/controller/core/workload_controller.go

```diff
@@ -142,45 +142,22 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 {
-		// Finish the workload
-		log.V(3).Info("Workload has Rejected admission checks, Finish with failure")
-		err := workload.UpdateStatus(ctx, r.client, &wl, kueue.WorkloadFinished,
-			metav1.ConditionTrue,
-			"AdmissionChecksRejected",
-			fmt.Sprintf("Admission checks %v are rejected", rejectedChecks),
-			constants.KueueName)
-
-		if err == nil {
-			for _, owner := range wl.OwnerReferences {
-				uowner := unstructured.Unstructured{}
-				uowner.SetKind(owner.Kind)
-				uowner.SetAPIVersion(owner.APIVersion)
-				uowner.SetName(owner.Name)
-				uowner.SetNamespace(wl.Namespace)
-				uowner.SetUID(owner.UID)
-				r.recorder.Eventf(&uowner, corev1.EventTypeNormal, "WorkloadFinished", "Admission checks %v are rejected", rejectedChecks)
-			}
-		}
-
-		return ctrl.Result{}, err
-	}
-
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
 		if updated, err := r.reconcileSyncAdmissionChecks(ctx, &wl, cqName); updated || err != nil {
 			return ctrl.Result{}, err
 		}
 	}
 
-	if workload.SyncAdmittedCondition(&wl) {
+	// If the workload is admitted, updating the status here will set the Admitted condition to
```

Comment:
```suggestion
	// If the workload is admitted, updating the status here would set the Admitted condition to
```
nit: it is a hypothetical if the implementation was differerent